db/batch/batch_impl.go

```diff
@@ -228,10 +228,8 @@ func (cb *cachedBatch) ClearAndUnlock() {
 	cb.kvStoreBatch.Clear()
 	// clear all saved snapshots
 	cb.tag = 0
-	cb.batchShots = nil
-	cb.cacheShots = nil
-	cb.batchShots = make([]int, 0)
-	cb.cacheShots = make([]KVStoreCache, 0)
+	cb.batchShots = cb.batchShots[:0]
+	cb.cacheShots = cb.cacheShots[:0]
```

Comment:
this will reuse the underlying memory, better than set to nil (discard current) and `make` a new slice